phconvert/pqreader.py

```diff
@@ -607,6 +752,40 @@ def process_t3records(t3records, time_bit=10, dtime_bit=15,
         ovcfunc = _correct_overflow
     ovcfunc(timestamps, detectors, overflow_ch, overflow)
     return detectors, timestamps, nanotimes
+def process_t3records_t3rfile(t3records, reserved=1, valid=1, time_bit=12, dtime_bit=16,
+                      ch_bit=2, special_bit=False):
+    """ For processing file.t3r format
+    time_bit: nanotimes
+    dtime_bit: TimeTag
+    if valid==1 the Data == Channel
+    else Data = Overflow[1], Reserved[8], Marker[3]
+    """
+    if special_bit:
+        ch_bit += 1
+    assert ch_bit <= 8
+    assert time_bit <= 16
+    assert time_bit+reserved+valid+dtime_bit+ch_bit == 32
+    
+    detectors = np.bitwise_and(
+        np.right_shift(t3records, time_bit+dtime_bit+reserved+valid), 2**ch_bit-1).astype('uint8')
+    nanotimes = np.bitwise_and(
+    np.right_shift(t3records, dtime_bit), 2**time_bit-1).astype('uint16')
```

Comment:
please add indentation on this line (continuation lines require indentation)

Reply:
done!